File: 242-valid-anagram/242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        
        if len(s) != len(t): 
            return False
        
        alpha = [0] * 128
        
        for c in s:
            alpha[ord(c)] += 1
        
        for c in t:
            if alpha[ord(c)] == 0:
                return False
            else:
                alpha[ord(c)] -= 1
        
        return True

File: 242-valid-anagram/242-valid-anagram.cpp
#define LEN 26
class Solution {
private:
    bool check_identical(vector<int>& sv, vector<int>& tv) {
        for (int i = 0; i < LEN; i++) {
            if (sv[i] != tv[i]) return false;
        }
        return true;
    }
public:
    bool isAnagram(string s, string t) {
        vector<int> sv(LEN, 0), tv(LEN, 0);
        for (auto& c : s) sv[c - 'a']++;
        for (auto& c : t) tv[c - 'a']++;
        return check_identical(sv, tv);
    }
};